In [105]:
# We first have to import pandas and numpy using import statement in python
import pandas as pd 
import numpy as np

In [106]:
#we have to import the csv file to be analysed by  uploading into the files
#create a data frame that can be read in the csv format
df=pd.DataFrame(pd.read_csv('//content/train (1).csv'))
#then print the first 5 part of the data set to catch a glimpse 
df.head

<bound method NDFrame.head of      PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0              1         0       3  ...   7.2500   NaN         S
1              2         1       1  ...  71.2833   C85         C
2              3         1       3  ...   7.9250   NaN         S
3              4         1       1  ...  53.1000  C123         S
4              5         0       3  ...   8.0500   NaN         S
..           ...       ...     ...  ...      ...   ...       ...
886          887         0       2  ...  13.0000   NaN         S
887          888         1       1  ...  30.0000   B42         S
888          889         0       3  ...  23.4500   NaN         S
889          890         1       1  ...  30.0000  C148         C
890          891         0       3  ...   7.7500   NaN         Q

[891 rows x 12 columns]>

In [107]:
#to determine the dimension of the data set 
df.shape


(891, 12)

In [108]:
#to count the total number of NaN values in the dataframe 
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [16]:
#from the information above , 
#cabin NaN value is over 40% of the total rows 
# Age NaN value is  over 15% of the rows. we need to drop the largest NaN value 
drop_col =df.isnull().sum()[df.isnull().sum()>(40/100 *df.shape[0])]
drop_col

Cabin    687
dtype: int64

In [18]:
#to determine the index of the dropped column and drop
drop_col.index

Index(['Cabin'], dtype='object')

In [109]:
#to see changes made and sum of NaN left 

df.isnull().sum()


PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [33]:
#to fill in NaN  in Age with values , 
#lets insert the mean of the age 
df.fillna(df.mean(),inplace=True)
df.isnull().sum()


PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       2
dtype: int64

In [34]:
#we still have 2 NaN in embarked, 
#decribe embark to see detail 
df['Embarked'].describe()

count     889
unique      3
top         S
freq      644
Name: Embarked, dtype: object

In [40]:
#most value in 'embarked' is in S, thus lets insertn s into the 2 left NaN
df['Embarked'].fillna ('S',inplace= True)

In [110]:
#correlation analysis to show the relationsip between the variables 
#to show how each variable move with the other 
df.corr

<bound method DataFrame.corr of      PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0              1         0       3  ...   7.2500   NaN         S
1              2         1       1  ...  71.2833   C85         C
2              3         1       3  ...   7.9250   NaN         S
3              4         1       1  ...  53.1000  C123         S
4              5         0       3  ...   8.0500   NaN         S
..           ...       ...     ...  ...      ...   ...       ...
886          887         0       2  ...  13.0000   NaN         S
887          888         1       1  ...  30.0000   B42         S
888          889         0       3  ...  23.4500   NaN         S
889          890         1       1  ...  30.0000  C148         C
890          891         0       3  ...   7.7500   NaN         Q

[891 rows x 12 columns]>

In [112]:
#lets create a new variable by merging the two 'parch' and Sibsp
df['Familysize'] = df['SibSp'] + df['Parch']


In [113]:
#Lets drop the two variables since a new one that can represent them had been created
df.drop(['SibSp', 'Parch'], axis =1 , inplace=True)


In [117]:
#correlate family size against surival rate
df['Familysize'].corr(df['Survived']) 
#below shows that there is a positive relationship between Family size and survival rate

0.016638989282745254

In [118]:
#correlate Age against surival rate
df['Age'].corr(df['Survived']) 
#below shows that there is a negative relationship between Age and Survival rate

-0.07722109457217768

In [119]:
#correlate pclass against surival rate
df['Pclass'].corr(df['Survived']) 
#below shows that there is a negative relationship between Pclass and Survival rate

-0.33848103596101475

In [120]:
#to check if being alone will affect survival rate
df['Alone']=[0 if df['Familysize'] [i]>0 else 1 for i in df.index]
df.head(15)

PassengerId  Survived  Pclass  ... Embarked Familysize  Alone
0             1         0       3  ...        S          1      0
1             2         1       1  ...        C          1      0
2             3         1       3  ...        S          0      1
3             4         1       1  ...        S          1      0
4             5         0       3  ...        S          0      1
5             6         0       3  ...        Q          0      1
6             7         0       1  ...        S          0      1
7             8         0       3  ...        S          4      0
8             9         1       3  ...        S          2      0
9            10         1       2  ...        C          1      0
10           11         1       3  ...        S          2      0
11           12         1       1  ...        S          0      1
12           13         0       3  ...        S          0      1
13           14         0       3  ...        S          6      0
14           15         0       3  ...        S          0      1

[15 rows x 12 columns]

In [121]:
#to calculate the mean of those who were alone '0' and survived 
#and those who are with their family "1'and survive
df.groupby(['Alone'])['Survived'].mean()

Alone
0    0.505650
1    0.303538
Name: Survived, dtype: float64

In [122]:
df['Alone'].corr(df['Survived'])
#there is a negative relationship between Being alone and the survival rate 

-0.20336708569989206

In [123]:
#the correlation between being alone and fares shows a  negatiive relationship
df['Alone'].corr (df['Fare'])

-0.2718323581061086

In [124]:
#to assign values 1 for female and 0 to male and to filter out the survival rate for each
df['Sex']=[0 if df['Sex'] [i]=='male' else 1 for i in df.index] #  1 for female , 0 for male
df.groupby(['Sex'])['Survived'].mean()

Sex
0    0.188908
1    0.742038
Name: Survived, dtype: float64

In [131]:
#to determine the survival relationship with the embarking station
df.groupby(['Embarked'])['Survived'].mean()

Embarked
C    0.553571
Q    0.389610
S    0.336957
Name: Survived, dtype: float64

In [133]:

#to describe the variable 'Survive '
df['Survived'].describe()

count    891.000000
mean       0.383838
std        0.486592
min        0.000000
25%        0.000000
50%        0.000000
75%        1.000000
max        1.000000
Name: Survived, dtype: float64

CONCLUSION:

1. An average of (38%) survived the titanic wreck.
2.The passengers who are on booard with their family have a higher chance of survival as they were prioritized and families help one another.
3.Children and young people  survive more as they were priotized above adults .
4.Passengers in the First class recieve more care as they were prioritized above other class , thus their chance of survival is higher .
5.Passengers who board with their family pay less fares compared to those travelling alone .
6.Female Passengers have higher chance of survival as they were priortized 
7.Passengers who boarded at Cherboug have higher chance of Survival.

We can recommend that anyone travelling should board at the Cherboug station, travel with their family and go through the First class (if they can afford)